app/src/lib/stats/stats-store.ts

```diff
@@ -1777,6 +1798,54 @@ export class StatsStore implements IStatsStore {
     }))
   }
 
+  // Generates the stat field name for the given PR review type and suffix.
+  private getStatFieldForRequestReviewState(
+    reviewType: IAPIPullRequestReview['state'],
+    suffix: PullRequestReviewStatFieldSuffix
+  ): PullRequestReviewStatField {
+    let infix: PullRequestReviewStatFieldInfix
+
+    if (reviewType === 'APPROVED') {
+      infix = 'Approved'
+    } else if (reviewType === 'CHANGES_REQUESTED') {
+      infix = 'ChangesRequested'
+    } else {
+      infix = 'Commented'
+    }
+
+    return `pullRequestReview${infix}${suffix}`
+  }
+
+  // Generic method to record stats related to Pull Request review notifications.
+  private recordPullRequestReviewStat(
+    reviewType: IAPIPullRequestReview['state'],
+    suffix: PullRequestReviewStatFieldSuffix
+  ) {
+    const statField = this.getStatFieldForRequestReviewState(
+      reviewType,
+      suffix
+    )
+    return this.updateDailyMeasures(m => ( { [statField]: m[statField] + 1 } as any ))
```

Comment:
This is the ugliest part, because I had to add `as any` here, otherwise TS expects us to include all fields in `PullRequestReviewStatField` in the new object.

It shouldn't be bad because `m[statField]` already makes sure `statField` is a key of `IDailyMeasures` (i.e., that `PullRequestReviewStatField extends keyof IDailyMeasures`), but it'd be nice if I could get rid of it!